File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProfit =0, minPrice= INT_MAX;
        int n=prices.size();
        // for (int i=0; i<n;i++){
        //     for(int j= i+1; j<n;j++){
        //         maxProfit= max(prices[j]-prices[i], maxProfit);
        //     }
        // }
        // return maxProfit;

        for(int i=0;i<n;i++){
            minPrice= min(minPrice, prices[i]);
            maxProfit= max(maxProfit, prices[i]- minPrice);
        }

        return maxProfit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let maxPrice = 0, minPrice = Infinity;
    for (let i = 0 ;i<prices.length; i++) {
        minPrice = Math.min(prices[i], minPrice)
        maxPrice = Math.max(maxPrice, prices[i] - minPrice)
    }

    return maxPrice;
};